src/ManiaTemplates/ManiaTemplateEngine.cs

```diff
@@ -280,4 +281,35 @@ public void RemoveManiaScript(string name)
 
         _maniaScripts.Remove(name);
     }
+
+    /// <summary>
+    /// Set a global variable that is passed to every component on render.
+    /// </summary>
+    public Task SetGlobalVariable(string key, object value)
```

Comment:
Perhaps its a better idea to use properties for these instead of methods? Seems unecessary.